rmw_connext_shared_cpp/include/rmw_connext_shared_cpp/wait.hpp

```diff
@@ -142,6 +228,23 @@ wait(
     }
   }
 
+  std::unordered_set<DDS::StatusCondition *> status_conditions;
+  // gather all status conditions with set masks
+  rmw_ret_t ret_code = __gather_event_conditions(events, status_conditions);
+  if (ret_code != RMW_RET_OK) {
+    return ret_code;
+  }
+  if (!status_conditions.empty()) {
+    // enable a status condition for each event
+    for (auto * status_condition : status_conditions) {
```

Comment:
Can't you just say `auto` here?

Reply:
Removed, the * is unnecessary here other than for readability.